### PR TITLE
gr-qtgui: Frequency sink freezes the running flowgraph

### PR DESCRIPTION
**Pull request by dl1ksv**
---

# Pull Request Details
With #5405 the segfault of freq_sink_c was fixed, but
if you enable the control panel and change the FFT size to 8192
while the flow graph is running, the whole flowgraph freezes.

## Description
This patch:

- fixes the freeze
- extends the fftsize to 32768 for freq_sink_c and freq_sink_f
- changes the entry for fftsize in the corresponding yml files from integer to enum
  and provides a drop down menu with those values that are provided by the gui
  in the running flowgraph.

The code provides code for setting the fftsize in form_menus.h and in freqcontrolpanel.cc
If you change some signal and slots you can do without the code from form_menus.h.
So this part is removed.

## Which blocks/areas does this affect?

freq_sink_c, freq_sink_f
## Testing Done

Flowgraph:
signal source -> freq_sink 
                      ->gui_sink
While running, change fftsize to each availabe size. No freeze at all.

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/master/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/master/CONTRIBUTING.md#dco-signed)
Signed-off-by: Volker Schroer <3470424+dl1ksv@users.noreply.github.com>

- [x] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/master/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Review by mormj — APPROVED** [review]
This definitely simplifies the freq sink block - I would hope that the dropdown is not problematic for anyone that is changing the parameter programmatically.  But the fixed range simplifies the logic.  This tests out as advertised.  LGTM

File: gr-qtgui/lib/freq_sink_c_impl.cc [hunk 1]
```diff
@@ -129,7 +129,7 @@ void freq_sink_c_impl::initialize()
     if (!d_name.empty())
         set_title(d_name);
 
-    set_output_multiple(d_fftsize);
+    set_output_multiple(32768);
```
Comment (mbr0wn) [inline gr-qtgui/lib/freq_sink_c_impl.cc:132]: Can we please have constants somewhere and reuse them?

File: gr-qtgui/grc/qtgui_freq_sink_x.block.yml [hunk 2]
```diff
@@ -20,8 +20,9 @@ parameters:
     hide: ${ ('none' if len(name) > 0 else 'part') }
 -   id: fftsize
     label: FFT Size
-    dtype: int
+    dtype: enum
```
Comment (mormj) [inline gr-qtgui/grc/qtgui_freq_sink_x.block.yml:23]: If you keep these as int, the dropdown persists, but the value can also be set to a variable

**Review by drmpeg — APPROVED** [review]
Tests good here.

**Comment by dl1ksv** [comment]
@mormj my patch nearly includes #5446, but here you can change the const's for the fft range and the values of the combobox are calculated. You don't have to edit the list.

**Comment by mormj** [comment]
Your solution here is much better - mine was a quick hack.

**Comment by drmpeg** [comment]
The error message is a little ugly. It should have a space after "be" and no spaces around the line feed.

Change:
```"FFT size must be>= {} and <= {}. \n So falling back to {}."```
to:
```"FFT size must be >= {} and <= {}.\nSo falling back to {}."
in both ```freq_sink_f_impl.cc and freq_sink_c.impl.cc

**Comment by drmpeg** [comment]
Also, the FFT size needs to be checked in the Qt GUI Sink. Values over 32768 cause this error on exit.

double free or corruption (out)

>>> Done (return code -6)

**Comment by drmpeg** [comment]
Is that Is that ${fftsize >= 32 and fftsize <= 32768} assert in the correct file? Shouldn't it be ```qtgui_sink_x.block.yml```?

**Comment by mormj** [comment]
The freq range was definitely change in freq sink, and this was made to match it:
https://github.com/gnuradio/gnuradio/pull/5442/files#diff-2d20922160ac6b37e729922683c1b8f2eec1dfdf2822b465ee6497e31ef5028aR144

**Comment by drmpeg** [comment]
I thought you were trying to fix my previous comment about the Qt Sink block.

**Comment by mormj** [comment]
No, I didn't touch that - was trying to get the last small pieces of the freq sink in.

**Comment by mormj** [comment]
It's worth doing but wanted to go ahead and get everything else merged first

**Comment by drmpeg** [comment]
Won't that assert make all that code dead?

**Comment by mormj** [comment]
Only if you're strictly in GRC, in which case it is nice to catch it earlier and get your red text.

**Comment by drmpeg** [comment]
Ok.

**Pull request merged**